File: backend/app/schemas/activity_logs.py
from typing import Optional
from pydantic import BaseModel, ConfigDict
import datetime


class ActivityLogCreate(BaseModel):
    action: str
    category: str
    target_title: Optional[str] = None
    details: Optional[str] = None


class ActivityLogResponse(BaseModel):
    id: str
    action: str
    category: str
    target_title: Optional[str] = None
    admin_email: str
    timestamp_str: Optional[str] = None
    details: Optional[str] = None
    created_at: Optional[datetime.datetime] = None

    model_config = ConfigDict(from_attributes=True)

File: src/api/activityLogs.ts
import { apiClient } from "./client"

export interface ActivityLog {
  id: string
  action: string
  category: "events" | "gallery" | "team" | "announcements" | "inquiries" | "applications" | "admins" | "settings"
  targetTitle?: string
  target_title?: string
  adminEmail: string
  admin_email?: string
  timestamp?: string
  timestamp_str?: string
  details?: string
  created_at?: string
}

export const activityLogsApi = {
  getAll: (limit = 50) =>
    apiClient<ActivityLog[]>(`/api/activity-logs?limit=${limit}`, { requireAuth: true }),
}
